pkg/envoy/accesslog/v3/commands.go

```diff
@@ -0,0 +1,196 @@
+	CMD_TRAILER          = "TRAILER"
+	CMD_DYNAMIC_METADATA = "DYNAMIC_METADATA"
+	CMD_FILTER_STATE     = "FILTER_STATE"
+	CMD_START_TIME       = "START_TIME"
+
+	// Commands that differ for HTTP and TCP log entries.
+
+	CMD_BYTES_RECEIVED        = "BYTES_RECEIVED"
+	CMD_BYTES_SENT            = "BYTES_SENT"
+	CMD_PROTOCOL              = "PROTOCOL"
+	CMD_RESPONSE_CODE         = "RESPONSE_CODE"
+	CMD_RESPONSE_CODE_DETAILS = "RESPONSE_CODE_DETAILS"
+	CMD_REQUEST_DURATION      = "REQUEST_DURATION"
+	CMD_RESPONSE_DURATION     = "RESPONSE_DURATION"
+	CMD_RESPONSE_TX_DURATION  = "RESPONSE_TX_DURATION"
+	CMD_GRPC_STATUS           = "GRPC_STATUS"
+
+	// Commands that are the same for HTTP and TCP log entries.
+
+	CMD_DURATION                                      = "DURATION"
+	CMD_RESPONSE_FLAGS                                = "RESPONSE_FLAGS"
+	CMD_UPSTREAM_HOST                                 = "UPSTREAM_HOST"
+	CMD_UPSTREAM_CLUSTER                              = "UPSTREAM_CLUSTER"
+	CMD_UPSTREAM_LOCAL_ADDRESS                        = "UPSTREAM_LOCAL_ADDRESS"
+	CMD_DOWNSTREAM_LOCAL_ADDRESS                      = "DOWNSTREAM_LOCAL_ADDRESS"
+	CMD_DOWNSTREAM_LOCAL_ADDRESS_WITHOUT_PORT         = "DOWNSTREAM_LOCAL_ADDRESS_WITHOUT_PORT"
+	CMD_DOWNSTREAM_REMOTE_ADDRESS                     = "DOWNSTREAM_REMOTE_ADDRESS"
+	CMD_DOWNSTREAM_REMOTE_ADDRESS_WITHOUT_PORT        = "DOWNSTREAM_REMOTE_ADDRESS_WITHOUT_PORT"
+	CMD_DOWNSTREAM_DIRECT_REMOTE_ADDRESS              = "DOWNSTREAM_DIRECT_REMOTE_ADDRESS"
+	CMD_DOWNSTREAM_DIRECT_REMOTE_ADDRESS_WITHOUT_PORT = "DOWNSTREAM_DIRECT_REMOTE_ADDRESS_WITHOUT_PORT"
+	CMD_REQUESTED_SERVER_NAME                         = "REQUESTED_SERVER_NAME"
+	CMD_ROUTE_NAME                                    = "ROUTE_NAME"
+	CMD_DOWNSTREAM_PEER_URI_SAN                       = "DOWNSTREAM_PEER_URI_SAN"
+	CMD_DOWNSTREAM_LOCAL_URI_SAN                      = "DOWNSTREAM_LOCAL_URI_SAN"
+	CMD_DOWNSTREAM_PEER_SUBJECT                       = "DOWNSTREAM_PEER_SUBJECT"
+	CMD_DOWNSTREAM_LOCAL_SUBJECT                      = "DOWNSTREAM_LOCAL_SUBJECT"
+	CMD_DOWNSTREAM_TLS_SESSION_ID                     = "DOWNSTREAM_TLS_SESSION_ID"
+	CMD_DOWNSTREAM_TLS_CIPHER                         = "DOWNSTREAM_TLS_CIPHER"
+	CMD_DOWNSTREAM_TLS_VERSION                        = "DOWNSTREAM_TLS_VERSION"
+	CMD_UPSTREAM_TRANSPORT_FAILURE_REASON             = "UPSTREAM_TRANSPORT_FAILURE_REASON"
+
+	// Commands that are the same for HTTP and TCP log entries
+	// but have no equivalent data in GrpcAccessLog.
+
+	CMD_DOWNSTREAM_PEER_FINGERPRINT_256 = "DOWNSTREAM_PEER_FINGERPRINT_256"
+	CMD_DOWNSTREAM_PEER_SERIAL          = "DOWNSTREAM_PEER_SERIAL"
+	CMD_DOWNSTREAM_PEER_ISSUER          = "DOWNSTREAM_PEER_ISSUER"
+	CMD_DOWNSTREAM_PEER_CERT            = "DOWNSTREAM_PEER_CERT"
+	CMD_DOWNSTREAM_PEER_CERT_V_START    = "DOWNSTREAM_PEER_CERT_V_START"
+	CMD_DOWNSTREAM_PEER_CERT_V_END      = "DOWNSTREAM_PEER_CERT_V_END"
+	CMD_HOSTNAME                        = "HOSTNAME"
```

Comment:
How dow e know there are not more coming from the new Envoy versions and v3 as a protocol?

Reply:
Actually, we don't even know if there aren't any new in V2 since it was implemented. Let me create a card to review it